src/test/java/com/es/phoneshop/web/ProductListPageServletTest.java

```diff
@@ -24,11 +25,14 @@
     private HttpServletResponse response;
     @Mock
     private RequestDispatcher requestDispatcher;
+    @Mock
+    private ServletConfig config;
 
     private ProductListPageServlet servlet = new ProductListPageServlet();
 
     @Before
-    public void setup(){
+    public void setup() throws ServletException {
+        servlet.init(config);
```

Comment:
Everything except the tested class should be mocked. In that case, product dao should be a @Mock field, injected into the tested instance via @InjectMocks, that also returns something meaningful (e.g. a singleton list of one @Mock product) from its findAll() method